src/types/either.rs

```diff
@@ -0,0 +1,254 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use actix_http::{Error, Response};
+use bytes::Bytes;
+use futures_util::{future::LocalBoxFuture, ready, FutureExt, TryFutureExt};
+use pin_project::pin_project;
+
+use crate::{dev, request::HttpRequest, FromRequest, Responder};
+
+/// Combines two different responder types into a single type
+///
+/// ```rust
+/// use actix_web::{Either, Error, HttpResponse};
+///
+/// type RegisterResult = Either<HttpResponse, Result<HttpResponse, Error>>;
+///
+/// fn index() -> RegisterResult {
+///     if is_a_variant() {
+///         // <- choose left variant
+///         Either::A(HttpResponse::BadRequest().body("Bad data"))
+///     } else {
+///         Either::B(
+///             // <- Right variant
+///             Ok(HttpResponse::Ok()
+///                 .content_type("text/html")
+///                 .body("Hello!"))
+///         )
+///     }
+/// }
+/// # fn is_a_variant() -> bool { true }
+/// # fn main() {}
+/// ```
+#[derive(Debug, PartialEq)]
+pub enum Either<A, B> {
+    /// First branch of the type
+    A(A),
```

Comment:
Why `A` over `Left` if the docs and test code use left / right terminology rather than A / B?

Reply:
its because this isnt a new type (i just moved the either parts out of responder.rs), it's in my head to rename the variants in next breaking release.